prov/efa/src/rdm/efa_rdm_pke_utils.c

```diff
@@ -71,9 +71,16 @@ ssize_t efa_rdm_pke_init_payload_from_ope(struct efa_rdm_pke *pke,
 		ret = efa_rdm_ep_use_p2p(pke->ep, iov_mr);
 		if (ret < 0)
 			return ret;
-		p2p_available = ret;
+		efa_can_access_memory = ret;
+	/**
+	 * efa can directly access memory if the buffer is host memory
+	 * and the pkt size (hdr + data) is within inline buf size.
+	 */
+	} else if (!efa_mr_is_hmem(iov_mr) &&
+	           payload_offset + data_size <= efa_rdm_ep_domain(pke->ep)->device->efa_attr.inline_buf_size){
```

Comment:
nit missing space between ) and {

Reply:
Also, consider reordering the if else checks:
```
if (!efa_mr_is_hmem(iov_mr) &&
	           payload_offset + data_size <= efa_rdm_ep_domain(pke->ep)->device->efa_attr.inline_buf_size) {
    ...
} else if (iov_mr) {
    ...
} else {
    ...
}
```

The current logic does not work for `FI_HMEM_SYSTEM`

Reply:
Updated!